system_metrics_collector/src/system_metrics_collector/cpu_data.hpp

```diff
@@ -15,36 +15,36 @@
 #ifndef SYSTEM_METRICS_COLLECTOR__CPU_DATA_HPP_
 #define SYSTEM_METRICS_COLLECTOR__CPU_DATA_HPP_
 
-#include <string>
+#include <array>
 #include <sstream>
+#include <string>
+
+/**
+ * Enum representing each item in a /proc/stat cpu line
+ */
+enum class ProcCpuStates
```

Comment:
Do you need to extend `int` here?
```c++
enum class ProcCpuStates : int
```

Reply:
It's initialized with a value, so I'm not sure why we need to extend. https://en.cppreference.com/w/cpp/language/enum

Reply:
TIL C++ well-defines `enum class` size based on the type of the first enum